spark/src/main/scala/org/apache/spark/sql/delta/files/DelayedCommitProtocol.scala

```diff
@@ -123,33 +126,49 @@ class DelayedCommitProtocol(
     }
   }
 
-  protected def parsePartitions(dir: String): Map[String, String] = {
-    // TODO: timezones?
+  protected def parsePartitions(dir: String,
+                                taskContext: TaskAttemptContext): Map[String, String] = {
     // TODO: enable validatePartitionColumns?
+    val utcTimestampPartitionValues = taskContext.getConfiguration.getBoolean(
+      DeltaSQLConf.UTC_TIMESTAMP_PARTITION_VALUES.key, true) &&
+      taskContext.isInstanceOf[PartitionedTaskAttemptContextImpl] &&
+      taskContext.asInstanceOf[PartitionedTaskAttemptContextImpl]
+        .partitionAttr.exists(attribute => attribute.dataType.isInstanceOf[TimestampType])
+    val sessionTimeZone = getZoneId(SQLConf.get.sessionLocalTimeZone)
+    val partitionCols: Map[String, DataType] = if (utcTimestampPartitionValues) {
+      taskContext.asInstanceOf[PartitionedTaskAttemptContextImpl]
+          .partitionAttr.groupBy(attr => attr.name)
+          .map(tup => (tup._1, tup._2.head.dataType))
+    } else {
+      Map.empty
+    }
+
     val dateFormatter = DateFormatter()
     val timestampFormatter =
-      TimestampFormatter(timestampPartitionPattern, java.util.TimeZone.getDefault)
+      TimestampFormatter(timestampPartitionPattern, sessionTimeZone)
     val parsedPartition =
       PartitionUtils
         .parsePartition(
           new Path(dir),
           typeInference = false,
           Set.empty,
-          Map.empty,
+          partitionCols,
```

Comment:
+100, I'll add some comments explaining all of this since it's def not an intuitive API to work with and in the end we probably want to get away from using it since we know the exact partition column data types for Delta.